cli/command_repository_set_parameters.go

```diff
@@ -56,6 +59,10 @@ func (c *commandRepositorySetParameters) setup(svc appServices, parent commandPa
 	cmd.Flag("epoch-delete-parallelism", "Epoch delete parallelism").IntVar(&c.epochDeleteParallelism)
 	cmd.Flag("epoch-checkpoint-frequency", "Checkpoint frequency").IntVar(&c.epochCheckpointFrequency)
 
+	cmd.Flag("metadata-compression", "Metadata compression algorithm").EnumVar(
+		&c.metadataCompression,
+		supportedCompressionAlgorithms()[1:]...) // Do not show `inherit` as a valid flag
```

Comment:
this seems fragile to me. Perhaps we can change the helper function so it doesn't return `"inherit"`, and we prepend it where needed:
```go

func supportedCompressionAlgorithms() []string {
	var res []string
	for name := range compression.ByName {
		res = append(res, string(name))
	}

	sort.Strings(res)

	return append([]string{"none"}, res...)
}

func withInherit(algos []string) []string {
	return append([]string{inheritPolicyString, }, algos...)
}
```

So usage in the policy compression flag becomes `withInherit(supportedCompressionAlgorithms())`:
```go
cmd.Flag("compression", "Compression algorithm").EnumVar(&c.policySetCompressionAlgorithm, withInherit(supportedCompressionAlgorithms())...)
```

In other words, we can build the slice we need instead of chopping off bits we don't need from a longer slice. Plus we won't need to assume anything about the position idx of `"inherit"`, or whether it can ever change. WDYT?

Reply:
Yeah, that makes sense